### PR TITLE
Restore EMF pen behaviour and increase compatability with WMF pen behaviour

### PR DESCRIPTION
**Pull request by hughbe**
@filipnavara

### CODE REVIEW

File: src/metafile.c [hunk 1]
```diff
@@ -494,20 +541,29 @@ gdip_metafile_CreatePenIndirect (MetafilePlayContext *context, DWORD style, DWOR
...
+		line_join = LineJoinRound;
+		break;
+	case PS_JOIN_BEVEL:
+		line_join = LineJoinBevel;
+		break;
+	case PS_JOIN_MITER:
+		line_join = LineJoinMiter;
+		break;
+	}
+	GdipSetPenLineJoin (pen, line_join);
```
Comment (filipnavara) [inline src/metafile.c]: I still thing this block should be wrapped in `if (dwPenStyle & PS_GEOMETRIC)`.

Comment (filipnavara) [reply src/metafile.c]: ...including the line caps (https://msdn.microsoft.com/en-us/library/windows/desktop/dd162711(v=vs.85).aspx).

**Comment by hughbe** [comment]
> In terms of avoiding code duplication maybe gdip_metafile_CreatePenIndirect can be implemented by calling gdip_metafile_ExtCreatePen?

Sort of, but then there would be subtly different behaviour to Windows GDI+. If you passed PS_GEOMETRIC to `CreatePen` nothing would happen currently (and this matches Windows GDI+), but if we implemented `createPen` in terms of `extCreatePen`, then the PS_GEOMETRIC flag would be listened to instead of ignored.

**Comment by filipnavara** [comment]
You cannot pass PS_GEOMETRIC to gdip_metafile_CreatePenIndirect in WMF because the values are read from 16-bit WORD and thus it can never be present in the data.

**Comment by filipnavara** [comment]
There are no test for EMR_CREATEPEN in EMF, but I expect it to respect the PS_GEOMETRIC style. Even if it was not, then it's simple to just mask it out in gdip_metafile_CreatePenIndirect.

**Comment by hughbe** [comment]
Derp you're absolutely right that's super clever! Fixed - good to have your eyes on this :D

**Pull request merged**